Error inserting Empatica S/N

If you try to insert 123456789 as S/N the app crashes and cannot be used anymore. The only way to use it again is to reinstall. 
11-18 14:07:27.529 6909-6909/? E/AndroidRuntime: FATAL EXCEPTION: main
                                                 Process: org.radarcns.android.android, PID: 6909
                                                 java.lang.RuntimeException: Unable to start service org.radarcns.empaticaE4.E4Service@5ff6337 with null: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.content.Intent.hasExtra(java.lang.String)' on a null object reference
                                                     at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:3008)
                                                     at android.app.ActivityThread.-wrap17(ActivityThread.java)
                                                     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1423)
                                                     at android.os.Handler.dispatchMessage(Handler.java:102)
                                                     at android.os.Looper.loop(Looper.java:148)
                                                     at android.app.ActivityThread.main(ActivityThread.java:5488)
                                                     at java.lang.reflect.Method.invoke(Native Method)
                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:746)
                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:636)
                                                  Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean android.content.Intent.hasExtra(java.lang.String)' on a null object reference
                                                     at org.radarcns.android.DeviceService.onInvocation(DeviceService.java:381)
                                                     at org.radarcns.empaticaE4.E4Service.onInvocation(E4Service.java:63)
                                                     at org.radarcns.android.DeviceService.onStartCommand(DeviceService.java:122)
                                                     at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:2991)
                                                     at android.app.ActivityThread.-wrap17(ActivityThread.java) 
                                                     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1423) 
                                                     at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                     at android.os.Looper.loop(Looper.java:148) 
                                                     at android.app.ActivityThread.main(ActivityThread.java:5488) 
                                                     at java.lang.reflect.Method.invoke(Native Method) 
                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:746) 
                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:636) 

--- a/app/src/main/java/org/radarcns/android/DeviceServiceConnection.java
+++ b/app/src/main/java/org/radarcns/android/DeviceServiceConnection.java
@@ -225,6 +225,9 @@ public class DeviceServiceConnection<S extends DeviceState>implements ServiceCon
      * @return
      */
     public boolean isAllowedDevice(String value) {
+        if ( getDeviceName() == null ) {
+            return false;
+        }
         return getDeviceName().contains(value);
     }
 

--- a/app/src/main/java/org/radarcns/empaticaE4/MainActivity.java
+++ b/app/src/main/java/org/radarcns/empaticaE4/MainActivity.java
@@ -615,7 +615,13 @@ public class MainActivity extends AppCompatActivity {
 
     public void updateServerRecordsSent(String keyNameTrigger, final Map<String,Integer> lastNumberOfRecordsSent )
     {
-        int numberOfRecordsTrigger = lastNumberOfRecordsSent.get(keyNameTrigger);
+        // Default to 0 if number of records cannot be retrieved
+        int numberOfRecordsTrigger;
+        try {
+            numberOfRecordsTrigger = lastNumberOfRecordsSent.get(keyNameTrigger);
+        } catch ( NullPointerException npe) {
+            numberOfRecordsTrigger = 0;
+        }
 
         // Condensing the message
         keyNameTrigger = keyNameTrigger.replaceFirst("_?android_?","");

